cadc-gms/src/main/java/org/opencadc/auth/StandardIdentityManager.java

```diff
@@ -0,0 +1,272 @@
+            ret.getPrincipals().add(p);
+            // TODO: this is sufficient for some purposes, but not for output using toDisplayString
+            // a real augment would require system credentials to implement
+        }
+        return ret;
+    }
+
+    @Override
+    public Object toOwner(Subject subject) {
+        // use NumericPrincipal aka OIDC sub for persistence
+        Set<NumericPrincipal> ps = subject.getPrincipals(NumericPrincipal.class);
+        if (ps.isEmpty()) {
+            return null;
+        }
+        return ps.iterator().next().getUUID().toString();
+    }
+
+    @Override
+    public String toDisplayString(Subject subject) {
+        // prefer HttpPrincipal aka OIDC preferred_username for string output, eg logging
+        Set<HttpPrincipal> ps = subject.getPrincipals(HttpPrincipal.class);
+        if (!ps.isEmpty()) {
+            return ps.iterator().next().getName(); // kind of ugh
+        }
+        
+        // try X509
+        Set<X500Principal> px = subject.getPrincipals(X500Principal.class);
+        if (!px.isEmpty()) {
+            return px.iterator().next().getName(); // kind of ugh
+        }
+        
+        return null;
+    }
+    
+    private void validateOidcAccessToken(Subject s) {
+        log.debug("validateOidcAccessToken - START");
+        Set<AuthorizationTokenPrincipal> rawTokens = s.getPrincipals(AuthorizationTokenPrincipal.class);
+        
+        log.debug("token issuer: "  + oidcIssuer + " rawTokens: " + rawTokens.size());
+        if (oidcIssuer != null && !rawTokens.isEmpty()) {
+            URL u = getUserEndpoint();
+            for (AuthorizationTokenPrincipal raw : rawTokens) {
+                String credentials = null;
+                String challengeType = null;
+
+                // parse header
+                log.debug("header key: " + raw.getHeaderKey());
+                log.debug("header val: " + raw.getHeaderValue());
+                if (AuthenticationUtil.AUTHORIZATION_HEADER.equalsIgnoreCase(raw.getHeaderKey())) {
+                    String[] tval = raw.getHeaderValue().split(" ");
+                    if (tval.length == 2) {
+                        challengeType = tval[0];
+                        credentials = tval[1];
+                    } else {
+                        throw new NotAuthenticatedException(challengeType, NotAuthenticatedException.AuthError.INVALID_REQUEST,
+                            "invalid authorization");
+                    }
+                } // else: some other challenge 
+                log.debug("challenge type: " + challengeType);
+                log.debug("credentials: " + credentials);
+                    
+                // validate
+                if (challengeType != null && credentials != null) {
+                    try {
+                        HttpGet get = new HttpGet(u, true);
+                        get.setRequestProperty("authorization", raw.getHeaderValue());
+                        get.prepare();
```

Comment:
Just want to confirm that prepare() will throw an exception for a non-200 response code

Reply:
yes, prepare() throws all the exceptions